server/aesdsocket.c

```diff
@@ -18,62 +18,322 @@ Reference: https://beej.us/guide/bgnet/html/
 #include <signal.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <time.h>
+#include <pthread.h>
 #include <sys/socket.h>
 #include <sys/types.h>
 #include <arpa/inet.h>
 #include <netinet/in.h>
 #include <linux/fs.h>
 #include <sys/stat.h>
 
-#define DEBUG 0
+
+
+#define DEBUG 1
 #define ASCII_NEWLINE 10
 #define PORT_NUMBER 9000
+#define INIT_BUF_SIZE 1024
+#define TIME_PERIOD 10
 
 char socket_file_path[50] = "/var/tmp/aesdsocketdata";
 
-int server_socket_fd;
-int client_socket_fd;
+bool kill_program = false;
+bool print_call_program = false;
+
 int socket_file_fd;
-char* rx_storage_buffer = NULL;
 
-void sig_handler(int signum)
+timer_t timer_id;
+
+pthread_mutex_t mutex;
+
+struct clean_up_data
 {
-    int status;
+    int server_soc_fd;
+    int client_soc_fd;
+    char* rx_storage_buff;
+};
+
+void *print_cal()
+{
+    struct timespec ts;
+    time_t t;
+    struct tm* temp;
+
+    while(!kill_program)
+    {
+        char buffer[INIT_BUF_SIZE] = {0};
+
+        if(0 !=clock_gettime(CLOCK_MONOTONIC, &ts))
+        {
+            perror("Clock get time failed: ");
+            break;
+        }
+        ts.tv_sec += TIME_PERIOD;
+
+        /* If signal received exit thread */
+        if(kill_program)
+        {
+            pthread_exit(NULL);
+        }
+
+        if(0 != clock_nanosleep(CLOCK_MONOTONIC, TIMER_ABSTIME, &ts, NULL))
+        {
+            perror("clock nanosleep failed: ");
+            break;
+        }
+        /* If signal received exit thread */
+        if(kill_program)
+        {
+            pthread_exit(NULL);
+        }
+        
+        t = time(NULL);
+        if (t == ((time_t) -1))
+        {
+            perror("time since epoch failed: ");
+            break;
+        }
+
+        temp = localtime(&t);
+        if (temp == NULL)
+        {
+            perror("Local time failed: ");
+            break;
+        }
+
+        int len = strftime(buffer, INIT_BUF_SIZE, "timestamp: %Y, %b, %d, %H:%M:%S\n", temp);
+        if(len == 0)
+        {
+            syslog(LOG_ERR, "Failed to get time");
+        }
+
+        if (0 != pthread_mutex_lock(&mutex)) {
+            perror("Mutex lock failed: ");
+            break;
+        }
+
+        if (-1 == write(socket_file_fd, buffer, len))
+        {
+            perror("write timestamp failed: ");
+        }
 
+        if (pthread_mutex_unlock(&mutex) != 0) {
+            perror("Mutex unlock failed: ");
+            break;
+        }
+    }
+    return NULL;
+}
+/*{
```

Comment:
@nihalthirunakarasu remove code rather than commenting out, let revision control save old versions for you.